### ISSUE TITLE
Region's play() does not fire pause/finish event after whole region is played

### ISSUE THREAD

**Issue by AndKiel** [katspaugh/wavesurfer.js#398]
I'm using wavesurfer together with html video tag and trying to achieve full sync of video with waveform.
Unfortunately when I use region's `play()` function, it's impossible to stop video from playing because it does not fire any events beside `region-play` (undocumented in readme, found in code) and `play`.
Could backend fire `pause` or `finish` after whole region is played?
Also when region starts playing it does not fire `seek` event when setting audio position to region's `start` value.

**Comment by katspaugh** [comment]
Hi Andrzej,

There's a region event called `region-out`. Both `region-play` and `region-out` events are listed in the README under the Region events section.

**Comment by AndKiel** [comment]
`region-play` is not listed in README.md file.
But `region-out` works for me, thank you very much. I assumed that `region-out` will fire even with normal playback, not only with `region.play()` function, my bad.

**Comment by AndKiel** [comment]
`region-play` is *not* listed in README.
And `region-out` does not work for me. It fires even when normal playback leaves region, not only when clicked region finished playing (`region.play()` plays specific part from beginning to end and after it finished there is no `pause` event). So right now for workaround I would probably have to bind my `region-out` callback on `region-click` and unbind on `region-out`.

**Comment by katspaugh** [comment]
You are right, thanks for clarifying. So the following should be done:

 * [ ] Region should fire the global `seek` event when it starts playing from start to end.
 * [ ] Region should fire `region-pause` when it finishes playing from start to end.
 * [ ] Document `region-play` in the README. Also make it clearer that region events correspond to global events prefixed with `region-`.

**Comment by AndKiel** [comment]
Yes.
But probably backend* should fire global events instead of region having its own.
Also it appears that when we use wavesurfer's `play` after region's `play`, it rewinds back to the region's start time and plays from there (shouldn't it start playing when it last stopped?).

*MediaElement and WebAudio - they are the ones with `seekTo`, `play` and `pause` not firing events, region just uses wavesurfer's `play` which uses backend's `play` (so it's probably impossible for region to have its own `region-pause` event, but global `play` and `pause` should be fine).

**Comment by kjantzer** [comment]
I'm looking for this feature as well. Seems inconsistent that when the region stops/pauses that the global `pause` event is not fired since the audio is indeed pausing.

**Comment by katspaugh** [comment]
I'll raise the priority of this issue. A PR would also be welcomed.

**Comment by kjantzer** [comment]
I was able to fix `region-out` not being fired when playing a specific region. The problem was the `audioprocess` time was milliseconds under the official end time of the region.

```javascript
// line 237 in regions plugin
my.end <= time // ex: 20 <= 19.998345121 would be the last audioprocess
```
 
I rounded the time to the nearest thousandth ms and now `region-out` fires

```javascript
if (!my.firedOut && my.firedIn && my.end <= Math.round(time*100)/100 ) {
```

**Comment by katspaugh** [comment]
Solved in #473 + d44c0464d5c79e9510df866f5e3ed21101408cad.
